app/jobs/allocation_job.rb

```diff
@@ -0,0 +1,34 @@
+class AllocationJob < ApplicationJob
+  queue_as :default
+
+  def perform(allocation_batch_job)
+    school = allocation_batch_job.school
+    order_state = allocation_batch_job.order_state
+    current_allocation = school.std_device_allocation
+
+    current_allocation_value = current_allocation&.allocation || 0
```

Comment:
```suggestion
    current_allocation_value = current_allocation&.raw_allocation || 0
```

`allocation` and `cap` can be from a virtual pool or a singular pool. In this instance I think we'll want to raise relative to the `raw_allocation`. Same for `cap` below.

Reply:
nice spot, done